play.pokemonshowdown.com/src/battle.ts

```diff
@@ -3442,6 +3444,7 @@ export class Battle {
 				this.messageFadeTime = 40;
 				this.isBlitz = true;
 			}
+			if (ruleName === 'Exact HP Mod') this.reportExactHP = true;
```

Comment:
why are you making a new property instead of checking `this.rules['Exact HP Mod']`

Reply:
was primarily was looking at something else was implemented and decided to mirror, if it should be checked that way then sure

Reply:
I left blitz and species clause when I added the `rules` object just to minimize refactoring at the time, but I think any new code that needs rules should just use the `rules` object if the only thing that needs to be checked is if the rule is in play.